templates/elixir/pkgs/default.nix

```diff
@@ -0,0 +1,41 @@
+{ lib, beam, ...}:
+
+let
+  # beam.interpreters.erlangR23 is available if you need a particular version
+  packages = beam.packagesWith beam.interpreters.erlang;
+
+  pname = "my-package";
+  version = "0.0.0";
+
+  src = ../.;
+
+  MIX_ENV="prod";
+  # flox will create a "fixed output derivation" based on
+  # the total package of fetched mix dependencies
+  mixFodDeps = packages.fetchMixDeps {
+    pname = "mix-deps-${pname}";
+    inherit src version;
+    # flox will complain and tell you the right sha256 value to replace this with
+    sha256 = lib.fakeSha256;
+    #sha256 = "";
+  };
+
+  configurePhase = ''
+    runHook preConfigure
+    mix deps.compile --no-deps-check
+    runHook postConfigure
+  '';
+
+in packages.mixRelease {
+  inherit src pname version mixFodDeps MIX_ENV;
```

Comment:
Is there a reason you're doing a `let...in` and `inherit` instead of just defining things here? I think that might be slightly more confusing for the non-Nix user

Reply:
@mkenigs I am not sure how what you describe would work in a nix expression exactly?

Reply:
I just meant move
```
  pname = "my-package";
  version = "0.0.0";

  src = ../.;

  MIX_ENV="prod";
  # flox will create a "fixed output derivation" based on
  # the total package of fetched mix dependencies
  mixFodDeps = packages.fetchMixDeps {
    pname = "mix-deps-${pname}";
    inherit src version;
    # flox will complain and tell you the right sha256 value to replace this with
    sha256 = lib.fakeSha256;
    #sha256 = "";
  };
```
inside the `packages.mixRelease`, but maybe I read over something

Reply:
@mkenigs I wasn't sure how to make your idea work with calling for instance `packages = beam.packagesWith beam.interpreters.erlang;`

Reply:
I am not sure I can change this to the way you are thinking and have it actually work. Not sure I can drop the `let/in` but if someone more nix knowledgeable than me has ideas, I am not opposed. Just don't know how to make that work

Reply:
Right that would have to stay outside of `packages.mixRelease`, I meant the other attributes

Reply:
I was able to move the `MIX_ENV` into this context. The rest are also used by the fetchMixDeps so I needed to retain them the way they are

Reply:
Makes sense sorry I just read over src/version/pname in mixFodDeps 😳

Reply:
@mkenigs in order to actually have this nix expression build, I had to still had to inherit in `fetchMixDeps` even if I moved these other vars into a recursed call to mixRelease

```nix
{ lib, beam, ...}:

let
  # beam.interpreters.erlangR23 is available if you need a particular version
  packages = beam.packagesWith beam.interpreters.erlang;
in packages.mixRelease rec {
  MIX_ENV="prod";
 
  pname = "my-package";
  version = "0.0.0";

  src = ../.;
  
  # flox will create a "fixed output derivation" based on 
  # the total package of fetched mix dependencies 
  mixFodDeps = packages.fetchMixDeps {
    inherit version src;
    pname = "mix-deps-${pname}";
    # nix will complain and tell you the right value to replace this with
    #sha256 = lib.fakeSha256;
    sha256 = "sha256-CYOhCqyuH2PtI7dAED+gqn1IJRbfYZZlSeG1wWtwJFs=";
    # if you have build time environment variables add them here
    #MY_VAR="value";
  };

  #postBuild = ''

    # for phoenix framework you can uncomment the lines below
    # for external task you need a workaround for the no deps check flag
    # https://github.com/phoenixframework/phoenix/issues/2690
    #mix do deps.loadpaths --no-deps-check, phx.digest
    #mix phx.digest --no-deps-check
   # mix do deps.loadpaths --no-deps-check
  #'';
}
```
I am not seeing a massive payoff in code readability/simplicity and clarity so I am going to push this and let's merge it

Reply:
Yep go ahead and merge!